elyra/pipeline/registry.py

```diff
@@ -0,0 +1,483 @@
+            if component_json is None:
+                continue
+            components['categories'].append(component_json)
+
+        return components
+
+    def get_properties(self, processor_type, component_id):
+        """
+        Return the properties JOSN for a given component.
+        """
+        default_components = ["notebooks", "python-script", "r-script"]
+
+        parser = self._get_parser(processor_type)
+
+        if parser._type == "local" or component_id in default_components:
+            properties = parser.get_common_config('properties')
+        else:
+            component = parser.return_component_if_exists(component_id)
+            if component is None:
+                raise ValueError(f"Component with ID {component_id} not found.")
+
+            reader = self._get_reader(component)
+
+            component_path = component['path'][reader._type]
+            component_body, _ = reader.get_component_body(component_path)
+            properties = parser.parse_component_properties(component_body)
+
+        return properties
+
+    def add_component(self, processor_type, request_body):
+        """
+        Add a component based on the provided definition. Definition will be provided in POST body
+        in the format {"name": "desired_name", path": {"file/url": "filepath/urlpath"}}.
+        """
+
+        parser = self._get_parser(processor_type)
+        parser.add_component(request_body)  # Maybe make this async to prevent reading issues in get_all_components()
+
+        components = self.get_all_components(None, parser._type)
+        return components
+
+    def get_component_execution_details(self, processor_type, component_id):
+        """
+        Returns the implementation details of a given component.
+        """
+        parser = self._get_parser(processor_type)
+        assert parser._type != "local"  # Local components should not have execution details
+
+        component = parser.return_component_if_exists(component_id)
+
+        reader = self._get_reader(component)
+        component_path = component['path'][reader._type]
+        component_body = reader.get_component_body(component_path)
+
+        execution_instructions = parser.parse_component_execution_instructions(component_body)
+        return execution_instructions
+
+    def _get_reader(self, component):
+        """
+        Find the proper reader based on the given registry component.
+        """
+
+        try:
+            component_type = list(component['path'].keys())[0]
+            return self.readers.get(component_type)
+        except Exception:
+            raise ValueError("Unsupported registry type.")
+
+    def _get_parser(self, processor_type: str):
+        """
+        Find the proper parser based on processor type.
+        """
+
+        try:
+            return self.parsers.get(processor_type)
+        except Exception:
+            raise ValueError(f"Unsupported processor type: {processor_type}")
```

Comment:
Since this is using `get()`, no exception will be thrown and `None` will be returned if `processor_type` is not in the dictionary.  I think you want `self.parsers[processor_type]` which will result in a `KeyError` being raised - which probably should be what is referenced in the `except` clause.

Reply:
Ah interesting! Thanks for pointing that out